storage/core/src/main/java/com/hortonworks/registries/storage/transaction/TransactionEventListener.java

```diff
@@ -86,9 +105,24 @@ public void onEvent(RequestEvent event) {
 
                 // get property whether to have unitOfWork with DB default transaction by default
                 useTransactionForUnitOfWork =
-                        unitOfWork.isPresent() ? unitOfWork.get().transactional() : runWithTxnIfNotConfigured;
-                TransactionIsolation transactionIsolation = unitOfWork.map(UnitOfWork::transactionIsolation)
-                                                                      .orElse(TransactionIsolation.DEFAULT);
+                        unitOfWork.map(UnitOfWork::transactional).orElse(runWithTxnIfNotConfigured);
+
+                TransactionIsolation transactionIsolation = null;
+
+                if (unitOfWork.isPresent()) {
```

Comment:
This block can be simplified to `Optional.map().orElse()` lambda style. I lean to this style but I am fine with either ways.

```
TransactionIsolation transactionIsolation =
        unitOfWork.map(x -> {
            TransactionIsolation result = x.transactionIsolation();
            if (result == TransactionIsolation.APPLICATION_DEFAULT) {
                result = defaultTransactionIsolation == null ? TransactionIsolation.DATABASE_SENSITIVE : defaultTransactionIsolation;
            }
            return result;
        }
).orElse(TransactionIsolation.DATABASE_SENSITIVE);
```